internal/agent/agent.go

```diff
@@ -82,63 +87,98 @@ func (a *Agent) sendMetrics() error {
-        {"lookups", float64(stats.Lookups)},
-        {"m-cache-inuse", float64(stats.MCacheInuse)},
-        {"m-cache-sys", float64(stats.MCacheSys)},
-        {"m-span-inuse", float64(stats.MSpanInuse)},
-        {"m-span-sys", float64(stats.MSpanSys)},
-        {"mallocs", float64(stats.Mallocs)},
-        {"next-gc", float64(stats.NextGC)},
-        {"num-forced-gc", float64(stats.NumForcedGC)},
-        {"num-gc", float64(stats.NumGC)},
-        {"other-sys", float64(stats.OtherSys)},
-        {"pause-total-ns", float64(stats.PauseTotalNs)},
-        {"stack-inuse", float64(stats.StackInuse)},
-        {"stack-sys", float64(stats.StackSys)},
-        {"sys", float64(stats.Sys)},
-        {"total-alloc", float64(stats.TotalAlloc)},
+        {"RandomValue", a.randValue},
+        {"Alloc", float64(stats.Alloc)},
+        {"BuckHashSys", float64(stats.BuckHashSys)},
+        {"Frees", float64(stats.Frees)},
+        {"GCCPUFraction", float64(stats.GCCPUFraction)},
+        {"GCSys", float64(stats.GCSys)},
+        {"HeapAlloc", float64(stats.HeapAlloc)},
+        {"HeapIdle", float64(stats.HeapIdle)},
+        {"HeapInuse", float64(stats.HeapInuse)},
+        {"HeapObjects", float64(stats.HeapObjects)},
+        {"HeapReleased", float64(stats.HeapReleased)},
+        {"HeapSys", float64(stats.HeapSys)},
+        {"LastGC", float64(stats.LastGC)},
+        {"Lookups", float64(stats.Lookups)},
+        {"MCacheInuse", float64(stats.MCacheInuse)},
+        {"MCacheSys", float64(stats.MCacheSys)},
+        {"MSpanInuse", float64(stats.MSpanInuse)},
+        {"MSpanSys", float64(stats.MSpanSys)},
+        {"Mallocs", float64(stats.Mallocs)},
+        {"NextGC", float64(stats.NextGC)},
+        {"NumForcedGC", float64(stats.NumForcedGC)},
+        {"NumGC", float64(stats.NumGC)},
+        {"OtherSys", float64(stats.OtherSys)},
+        {"PauseTotalNs", float64(stats.PauseTotalNs)},
+        {"StackInuse", float64(stats.StackInuse)},
+        {"StackSys", float64(stats.StackSys)},
+        {"Sys", float64(stats.Sys)},
+        {"TotalAlloc", float64(stats.TotalAlloc)},
     }
 
     for _, m := range gaugeMetrics {
-        err := a.sendMetric("gauge", m.name, fmt.Sprintf("%.2f", m.value))
+        err := a.sendMetric(models.Gauge, m.name, m.value)
         if err != nil {
             return err
         }
     }
 
-    return a.sendMetric("counter", "poll-count", strconv.Itoa(a.pollCount))
+    return a.sendMetric(models.Counter, "PollCount", int64(a.pollCount))
 }
 
-func (a *Agent) sendMetric(t string, name string, value string) error {
+func (a *Agent) sendMetric(t string, name string, value interface{}) error {
     if len(a.addr) == 0 {
         return errors.New("host must be configured")
     }
 
+    metric := models.Metrics{
+        ID:    name,
+        MType: t,
+    }
+
+    switch t {
+    case models.Counter:
+        v, ok := value.(int64)
+        if !ok {
+            return errTypeCast
```

Comment:
К ошибка из сторонних пактеов лучше добавлять контекст, объясняющий, что именно пошло не так:

```suggestion
            return fmt.Errorf("не удалось привести значение %v к типу int64: %w", err)
```

Обрати внимание, что текст аналогичной ошибки чуть дальше будет другим